### PR TITLE
fix(runtime): revert to Vim's color scheme in bundled color schemes

### PR DESCRIPTION
**Pull request by echasnovski**
Problem: Bundled color schemes use `hi clear` and define only Vim's
  highlight groups. This results into Neovim specific highlight groups
  use definitions from its default color scheme, which are not always
  linked to a Vim's highlight group.

Solution: Restore links to Vim's highlight groups which were present
  before Neovim default color scheme update.

------

I added `:source $VIMRUNTIME/colors/vim.lua` recipe alongside `colorscheme vim` as it is slightly different (at least, doesn't trigger `ColorScheme` event).

------

Related to #26369.

### CODE REVIEW

**Comment by echasnovski** [comment]
CI fails don't look related to me. Not sure why Windows one did not move past where it stopped. I believe `:source $VIMRUNTIME/colors/vim.vim` should still work on Windows, right?

**Comment by echasnovski** [comment]
As it seems to be better to rewrite 'colors/vim.vim' into Lua (to make a distinction between runtime files coming from Vim and Neovim), I converted this to draft. Will update to `source $VIMRUNTIME/colors/vim.lua` once it is rewritten.

**Comment by echasnovski** [comment]
So this should be updated after #26658 is merged. If that PR is closed without merged, then this PR is good to go as is.

**Comment by echasnovski** [comment]
Updated to account for the merged #26658.

**Review by clason — APPROVED** [review]

File: runtime/colors/blue.vim [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/blue.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/darkblue.vim [hunk 2]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/darkblue.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/delek.vim [hunk 3]
```diff
@@ -10,7 +10,8 @@
 
 set background=light
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/delek.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/desert.vim [hunk 4]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/desert.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/elflord.vim [hunk 5]
```diff
@@ -9,7 +9,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/elflord.vim:13]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/evening.vim [hunk 6]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/evening.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/habamax.vim [hunk 7]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/habamax.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/industry.vim [hunk 8]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/industry.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/koehler.vim [hunk 9]
```diff
@@ -9,7 +9,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/koehler.vim:13]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/lunaperche.vim [hunk 10]
```diff
@@ -8,7 +8,8 @@
 
 " Generated by Colortemplate v2.2.3
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/lunaperche.vim:12]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/morning.vim [hunk 11]
```diff
@@ -10,7 +10,8 @@
 
 set background=light
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/morning.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/murphy.vim [hunk 12]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/murphy.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/pablo.vim [hunk 13]
```diff
@@ -9,7 +9,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/pablo.vim:13]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/peachpuff.vim [hunk 14]
```diff
@@ -10,7 +10,8 @@
 
 set background=light
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/peachpuff.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/quiet.vim [hunk 15]
```diff
@@ -8,7 +8,8 @@
 
 " Generated by Colortemplate v2.2.3
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/quiet.vim:12]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/retrobox.vim [hunk 16]
```diff
@@ -8,7 +8,8 @@
 
 " Generated by Colortemplate v2.2.3
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/retrobox.vim:12]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/ron.vim [hunk 17]
```diff
@@ -9,7 +9,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/ron.vim:13]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/shine.vim [hunk 18]
```diff
@@ -10,7 +10,8 @@
 
 set background=light
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/shine.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/slate.vim [hunk 19]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/slate.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/sorbet.vim [hunk 20]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/sorbet.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/torte.vim [hunk 21]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/torte.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/wildcharm.vim [hunk 22]
```diff
@@ -8,7 +8,8 @@
 
 " Generated by Colortemplate v2.2.3
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/wildcharm.vim:12]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/zaibatsu.vim [hunk 23]
```diff
@@ -10,7 +10,8 @@
 
 set background=dark
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/zaibatsu.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

File: runtime/colors/zellner.vim [hunk 24]
```diff
@@ -10,7 +10,8 @@
 
 set background=light
 
-hi clear
+" hi clear
+source $VIMRUNTIME/colors/vim.lua " Neovim: revert to Vim default color scheme
```
Comment (zeertzjq) [inline runtime/colors/zellner.vim:14]:
```suggestion
source $VIMRUNTIME/colors/vim.lua " Nvim: revert to Vim default color scheme
```

**Pull request merged**